File: 브론즈 1 (5문제)/8595_히든 넘버.c
// 참고 링크 : https://moon-gd.tistory.com/36
#define _CRT_SECURE_NO_WARNINGS
#include <stdio.h>
char sentence[5000000];

int main() {
	long long sum = 0;
	int hidden_number = 0;
	int length = 0;
	char temp_c;

	scanf("%d", &length);
	scanf("%c", &temp_c);
	scanf("%s", sentence);
	for (int i = 0; i < length; i++) {
		temp_c = sentence[i];
		// 숫자인 경우
		if (0 <= temp_c - '0' && temp_c - '0' <= 9) {
			if (hidden_number == 0) {
				hidden_number += temp_c - '0';
			}
			else {
				hidden_number *= 10;
				hidden_number += temp_c - '0';
			}
		}
		else {
			sum += hidden_number;
			hidden_number = 0;
		}
	}

	// 마지막에 숫자가 들어온 경우 hidden_number 값이 아직 남아있기 때문에 더해준다
	sum += hidden_number;

	// 합 출력
	printf("%lld", sum);

	return 0;
}

File: 브론즈 1 (5문제)/8595_히든 넘버.py
# 참고 링크 : https://moon-gd.tistory.com/36
length = int(input())
sentence = input().lower().rstrip()

alphabet = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i',
            'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r',
            's', 't', 'u', 'v', 'w', 'x', 'y', 'z']

for c in alphabet:
    sentence = sentence.replace(c, ' ')

sentence = list(map(int, sentence.split()))

print(sum(sentence))